node/pkg/driver/device_connectivity/device_connectivity_helper_scsigeneric.go

```diff
@@ -169,19 +218,19 @@ func (r OsDeviceConnectivityHelperScsiGeneric) GetMpathDevice(volumeId string) (
 	if dmPath == "" {
 		return "", &MultipathDeviceNotFoundForVolumeError{volumeId}
 	}
-
-	SgInqWwn, err := r.Helper.GetWwnByScsiInq(dmPath)
+	_, err = r.Helper.IsMpathHasRightVolumeId(dmPath, volumeUuidLower, volumeNguid)
 	if err != nil {
 		return "", err
 	}
-	SgInqWwnLower := strings.ToLower(SgInqWwn)
-	if SgInqWwnLower != volumeUuidLower && SgInqWwnLower != volumeNguid {
-		// To make sure we found the right WWN, if not raise error instead of using wrong mpath
-		return "", &ErrorWrongDeviceFound{dmPath, volumeUuidLower, SgInqWwn}
-	}
 	return dmPath, nil
 }
 
+func (OsDeviceConnectivityHelperScsiGeneric) GetNguidFromVolumeId(volumeUuid string) (string, string) {
+	volumeUuidLower := strings.ToLower(volumeUuid)
+	volumeNguid := ConvertScsiIdToNguid(volumeUuidLower)
```

Comment:
```suggestion
	volumeNguid := convertScsiIdToNguid(volumeUuidLower)
```

Reply:
do we even still need `convertScsiIdToNguid` as a separate function? `GetNguidFromVolumeId` doesn't do much

Reply:
I think it's more readable that way with `convertScsiIdToNguid` function, because when you want to understand what `GetNguidFromVolumeId` does, you don't need to read all the content of `convertScsiIdToNguid` and just understand it from the name, but if you think we should remove this function and add the code in `GetNguidFromVolumeId` we can talk about

Reply:
and changed for now the `ConvertScsiIdToNguid` function name

Reply:
perhaps the `GetNguidFromVolumeId` name is the confusing part.
how about something like `GetVolumeIdVariations`, which would return a list (containing currently two items)?

Reply:
then functions such as `WaitForDmToExist` would get this list as a parameter (instead of two separate parameters)

Reply:
done

Reply:
what about my [last](https://github.com/IBM/ibm-block-csi-driver/pull/491#discussion_r873126207) [comments](https://github.com/IBM/ibm-block-csi-driver/pull/491#discussion_r873158182) in this conversation?